sthgetter/sthgetter.go

```diff
@@ -68,7 +72,72 @@ func getCheckStoreSTH(ctx context.Context, lc *client.LogClient, st storage.APIC
 		glog.Infof("%s: %s: error writing API Call %s: %s", l.URL, logStr, apiCall, err)
 	}
 
-	//TODO(katjoyce): Run checks on the received STH.
+	if sth == nil {
+		return
+	}
+
+	// Verify the STH.
+	errs := checkSTH(sth, apiCall.End, sv, l)
+	if len(errs) != 0 {
+		var b strings.Builder
+		fmt.Fprintf(&b, "STH verification errors for STH %v:", sth)
+		for _, e := range errs {
+			fmt.Fprintf(&b, "\n\t%v: %v,", reflect.TypeOf(e), e)
+		}
+		glog.Info(b.String())
+	}
+
+	// TODO(katjoyce): Store STH & errs.
+}
+
+func checkSTH(sth *ct.SignedTreeHead, receivedAt time.Time, sv *ct.SignatureVerifier, l *ctlog.Log) []error {
+	var errs []error
+	// Check that the STH signature verifies.
+	log.Printf("%s: %s: verifying STH signature...", l.URL, logStr)
```

Comment:
This function uses `log` instead of `glog`.

Reply:
nit: Could omit all of the `l.URL` and `logStr` arguments in this function's logging and just add them to the header you prepend in `getCheckStoreSTH()`. When you implement storing of the errors, you may not want those fields in the error message (presumably they'll be stored separately).

Reply:
oops - think this is because I started writing it before the switch to glog!  Fixed :)

You're right that I needed to prepend the `l.URL` and `logStr` to the logging of the errs in getCheckStoreSTH (which I have now done), but removing them from the logging in this function seems wrong - unless you are suggesting removing the logging altogether?  Which I also wouldn't want to do completely... although there is currently potentially a little bit of redundancy, but I'll remove that when I add storing the errors I think.  `l.URL` and `logStr` aren't put in the errors returned by this function at the moment, they're only used for logging.  Maybe I've misunderstood your point?

Reply:
Sorry, I misread the code a bit and thought the returned errors had `l.URL` and `logStr` prefixed. The changes you've made look good, thanks.